### DISCUSSION TITLE
Proxying prisma client with socksjs?

### DISCUSSION THREAD

**Discussion by li-dennis** [prisma/prisma#4163] [Q&A] [answered]
Hi,

Apologies if this has been asked before but I haven't been able to find any info. But as the title implies: how do I go about connecting to a database through a proxy? (The reason for this is that we do not want direct access to the DB instance which sits in a VPC separate from the server)

I'd like to setup a connection with something like socksjs, but I'm not sure how to go about overriding the prisma client connection behavior since it seems to expect a direct connection. Looking at the prisma docs I'm not really sure how I'd go about fiddling the behavior to do this.

eg normally in mysql i could do something like this (https://devcenter.heroku.com/articles/quotaguardstatic#socks5-proxy).

```
var mysql = require('mysql2'),
    url = require('url'),
    SocksConnection = require('socksjs');

var remote_options = {
    host:'your-database.eu-west-1.rds.amazonaws.com',
    port: 3306
};

var proxy = url.parse(process.env.QUOTAGUARDSTATIC_URL),
    auth = proxy.auth,
    username = auth.split(':')[0],
    pass = auth.split(':')[1];

var sock_options = {
    host: proxy.hostname,
    port: 1080,
    user: username,
    pass: pass
};

var sockConn = new SocksConnection(remote_options, sock_options);
var dbConnection = mysql.createConnection({
    user: 'dbuser',
    database: 'dbname',
    password: 'dbpassword',
    stream: sockConn
});
dbConnection.query('SELECT 1+1 as test1;', function(err, rows, fields) {
    if (err) throw err;

    console.log('Result: ', rows);
    sockConn.dispose();
});
dbConnection.end();

```

**Comment by ryands17** [comment, marked as answer]
@li-dennis 👋 
You can use SSH port forwarding in this case (I personally use this).

```
ssh -N -L 5432:password.rdsHost:5432 ec2-instance@ip-address -i keyfile.pem -v
```

And then I can use `localhost` as my DB host. Let me know if that helps :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
